### ISSUE TITLE
Warn user when they try to use gunicorns --log-level

### ISSUE THREAD

**Issue by bloodearnest** [canonical-ols/talisker#35]
Should explain DEBUGLOG mechanism

Perhaps consider changing the argparse parser message for --log-level and other things in talisker case?

**Comment by thomir** [comment]
I've just run into this, and it's hugely frustrating.

personally, in development I want to see the DEBUG level log messages in the console. Please could you consider making it so if I specify '--log-level debug' in the talisker options my debug messages make it to the console?

Cheers,

**Comment by bloodearnest** [comment]
Hey Thomi

tl;dr I think we should probably do this. Comes down to pythons "consenting adults" idea really, and I think not violating the principle of least surprise here is probably important.

The implementation is a little nuanced, so if you could answer some questions, that would help get it right. Honest feedback appreciated!

1) Just checking you are aware of the DEBUGLOG env var http://talisker.readthedocs.io/en/latest/logging.html#debug-logging. Thoughts on this?

2) Why specifically do you want debug output in the console rather than a file?

3) Do you set the DEVEL env var when doing local development? http://talisker.readthedocs.io/en/latest/logging.html#development. Did you know about this? Do you think it is useful/confusing/etc?

FWIW, never sending debug level logging to stderr is designed to prevent both information leakage and excessive volume in prod. We can relax this for sure in dev, but want to avoid making it easy to accidentally do this in production. E.g, one option would be to require DEVEL env var for it to work. Or we could just remove the restriction altogether iff --log-level=debug for gunicorn.

**Comment by thomir** [comment]
Hi Simon,

Answers below:

1) Yup - that's what I'm using at the moment, but it's a bit of a pain: I now have to have two terminals open: one where talisker is running, and another that's got `tail -f debuglog` running.

2) As hinted at above - I use the debug log level to show me what's going on from a developer perspective, which is really useful when running in a development context. Having it go to a separate file means I need two terminals, or some fancy redirection... 

Additionally, in a development context I don't care about persisting the logs: I won't ever refer to the debug log or the main log outside of the moment I'm in, so I really don't need them to be collecting in a file on disk.

3) Yup, we have `DEVEL=1` set in all our makefiles. I love the colourised output, and the auto-reloading (although that seems somewhat flaky - sometimes I need to kill & restart talisker in order to get it to load my new code, but it's not annoying enough for me to track down what's actually going on yet). 

I personally wouldn't mind if `DEVEL=1` additionally turned on debug logs, but I understand that might be annoying for some existing users.

I suggest just making --log-level=DEBUG work as expected: fold the debug level logs in to the rest of the logs that talisker prints to the terminal.


Cheers,

**Comment by bloodearnest** [comment]
What I ended up doing is combining this with an idea I'd had for a while, which is using sys.stderr.isatty() as a proxy for DEVEL=1 in the env.

So, now, without any special env vars, if you do --log-level=debug with gunicorn, you get exactly what you'd expect. And we are still protected in prod, where sys.stderr.isatty() is False, debug logs won't go to stderr. 

So, this worked out pretty well, IMO.
